### PR TITLE
fix stack overflow issue when using initializer_list

### PR DESCRIPTION
**Pull request by ghost**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix unit test `collapsing_sorted_stream`.

### Detailed description / Documentation draft:
Fix stack overflow issue in copy on write pointer when using initializer_list
The `create` function inside COWHelper calls itself when called with initializer_list

### CODE REVIEW

**Comment by KochetovNicolai** [comment]
It is not clear for me what problem was solved. Could you please explain it a little bit or add a test?

**Comment by huiwangcs** [comment]
hello @KochetovNicolai, when running dbms/src/DataStreams/tests/**collapsing_sorted_stream**.cpp, i've an exit code 11 and when debugging with lldb, i found the stack is full of calls on `static MutablePtr create(std::initializer_list<T> && arg) { return create(std::forward<std::initializer_list<T>>(arg)); }`

It seems that this method is calling itself recursively.

I'm running the program on MacOS 10.14.6 with clang 8.

**Comment by alexey-milovidov** [comment]
I have also faced this issue. It cannot happen in production, because this constructor is unused. I tried to fix it some time ago but faced some difficulties. Need to review the fix.

**Comment by huiwangcs** [comment]
Thanks for your feedback, @alexey-milovidov. Feel free to comment on the code. The fix is done together with @LeoErcolanelli who contributed to some features previsouly.

**Review by alexey-milovidov — APPROVED** [review]
The fix looks Ok (and previous code was looking like an obvious infinite recursion).

File: dbms/src/DataStreams/tests/collapsing_sorted_stream.cpp [hunk 1]
```diff
@@ -30,13 +30,13 @@ try
         ColumnWithTypeAndName column1;
         column1.name = "Sign";
         column1.type = std::make_shared<DataTypeInt8>();
-        column1.column = ColumnInt8::create({1, -1});
+        column1.column = ColumnInt8::create({static_cast<int8_t>(1), static_cast<int8_t>(-1)});
```
Comment (alexey-milovidov) [inline dbms/src/DataStreams/tests/collapsing_sorted_stream.cpp]: Why don't make the type of initializer list generic in ColumnVector and PODArray constructors? Let me try to do it in subsequent PR...

**Comment by huiwangcs** [comment]
@alexey-milovidov is it normal that the CI is not completely green? Do tests fail randomly?

**Comment by alexey-milovidov** [comment]
It isn't. Fails are not random.

**Pull request merged**